core/modules/qdisp/QueryResource.cc

```diff
@@ -44,46 +44,53 @@
 #include "qdisp/JobStatus.h"
 #include "qdisp/QueryRequest.h"
 #include "qdisp/QueryResource.h"
-#include "qdisp/ResponseRequester.h"
 
 namespace lsst {
 namespace qserv {
 namespace qdisp {
 
+QueryResource::QueryResource(std::shared_ptr<JobQuery> jobQuery)
+      // this char* must live as long as this object, so copy it on heap
+    : Resource(::strdup(jobQuery->getDescription().resource().path().c_str())),
+      _xrdSsiSession(NULL), _jobQuery(jobQuery) {
```

Comment:
Yes, got that.  I was asking if this strdup/free is still necessary (given that the JobQuery that gives it to us is going to outlive us?)